lib/scraper_test.rb

```diff
@@ -30,17 +31,42 @@ def install_tasks
           it 'should contain the expected data' do
             # TODO: Make this configurable
             Dir['test/data/*.yml'].each do |file|
-              yaml_data = YAML.load_file(file).to_h
-              url = yaml_data[:url]
-              class_to_test = Object.const_get(yaml_data[:class])
-              VCR.use_cassette(File.basename(url)) do
-                response = class_to_test.new(response: Scraped::Request.new(url: url).response)
-                response.to_h.must_equal yaml_data[:to_h]
+              VCR.use_cassette(File.basename(file)) do
+                instructions = Instructions.new(FilePath.new(file))
+                response = instructions.class_to_test
+                                       .new(response: Scraped::Request.new(url: instructions.url).response)
+                response.to_h.must_equal instructions.expected
               end
             end
           end
         end
       end
     end
   end
+
+  class Instructions
+    def initialize(filepath)
+      @filepath = filepath
+    end
+
+    def url
+      test_data[:url] or raise "No url provided by #{filepath.basename}."
```

Comment:
With the way you've implemented this, this `raise` doesn't actually work. However, as all you're doing in the tests is checking that you got a `StandardError`, that neatly disguises the error with your error! 

I suggest you adjust the tests to make sure that you're actually getting the error message you expect.